### PR TITLE
Fix undefined node error on save - part 2

### PR DESCRIPTION
**Pull request by t1m0thyj**
## Proposed changes

* Fixes document being removed from `openFiles` map prematurely when diff editor is closed
* Fixes node not found on save when favorited file is opened outside of favorites
* Adds error handling for case when file is saved and node cannot be found in tree

## Release Notes

Milestone: 2.15.3

Changelog: Fixed issue where saving changes to favorited data set or USS file could fail when it is opened outside of favorites

## Types of changes

What types of changes does your code introduce to Zowe Explorer?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Updates to Documentation or Tests (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This checklist will be used as reference for both the contributor and the reviewer_

- [ ] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/zowe-explorer-vscode/wiki/Best-Practices:-Contributor-Guidance) wiki
- [ ] PR title follows [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0-beta.2/)
- [ ] PR Description is included
- [ ] gif or screenshot is included if visual changes are made
- [ ] `yarn workspace vscode-extension-for-zowe vscode:prepublish` has been executed
- [ ] All checks have passed (DCO, Jenkins and Code Coverage)
- [ ] I have added unit test and it is passing
- [ ] I have added integration test and it is passing
- [ ] There is coverage for the code that I have added
- [ ] I have tested it manually and there are no regressions found
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any PR dependencies have been merged and published (if appropriate)

### CODE REVIEW

**Review by zFernand0 — COMMENTED** [review]
would it be ok to rebase this draft against `main`?
or were we planning on yet another patch release? :yum:

**Comment by t1m0thyj** [comment]
> would it be ok to rebase this draft against `main`? or were we planning on yet another patch release? 😋

I believe we plan another patch release, so I've added this PR to the 2.15.3 milestone 😁

File: packages/zowe-explorer/src/shared/utils.ts [hunk 1]
```diff
@@ -375,9 +379,13 @@ export async function compareFileContent(
     label?: string,
     profile?: imperative.IProfileLoaded
 ): Promise<void> {
+    node = node ?? TreeProviders.ds.openFiles?.[doc.uri.fsPath] ?? TreeProviders.uss.openFiles?.[doc.uri.fsPath];
```
Comment (zFernand0) [inline packages/zowe-explorer/src/shared/utils.ts:382]: Curious if we should also search for the node in the `uploadContent` method before we check for `== null`.

**Review by zFernand0 — APPROVED** [review]
Changes LGTM! 😋 
And I'm not getting any `cannot read property of undefined (reading ...)`

**Review by traeok — APPROVED** [review]
LGTM - thanks @t1m0thyj

**Review by JillieBeanSim — CHANGES_REQUESTED** [review]
Hey all, just adding the changes requested to ask about the Theia test on this branch that are broken?  Looks like only the first test passed and the rest fail.

**Comment by t1m0thyj** [comment]
> Hey all, just adding the changes requested to ask about the Theia test on this branch that are broken? Looks like only the first test passed and the rest fail.

@JillieBeanSim Good catch, in order to fix the Theia tests we'll need to cherry-pick some commits from https://github.com/zowe/zowe-explorer-vscode/pull/2838. Should I create a separate PR that contains just those commits? I don't want to be blocking other PRs going into maintenance with failing Theia tests 😋

**Review by JillieBeanSim — APPROVED** [review]

**Comment by JillieBeanSim** [comment]
broken Theia tests are not specific to the branch, breakages are already introduced to maintenance branch

**Comment by JillieBeanSim** [comment]
> > Hey all, just adding the changes requested to ask about the Theia test on this branch that are broken? Looks like only the first test passed and the rest fail.
> 
> @JillieBeanSim Good catch, in order to fix the Theia tests we'll need to cherry-pick some commits from #2838. Should I create a separate PR that contains just those commits? I don't want to be blocking other PRs going into maintenance branch with failing Theia tests 😋

Hey @t1m0thyj no worries, a separate PR for maintenance branch would be good and can link the open issue I created to track https://github.com/zowe/zowe-explorer-vscode/issues/2855 😄

File: packages/zowe-explorer/src/shared/utils.ts (hunk 1 above)
Comment (t1m0thyj) [reply packages/zowe-explorer/src/shared/utils.ts:382]: At the time `uploadContent` is called, we should have already handled the case where node is null or undefined in `saveFile` for data sets or `saveUSSFile` for USS files.

**Pull request merged**